sklearn/multiclass.py

```diff
@@ -409,6 +476,46 @@ def fit(self, X, y):
 
         return self
 
+    def partial_fit(self, X, y, classes=None):
+        """Partially fit underlying estimators 
+        
+        Should be used when memory is inefficient to train all data. Chunks 
+        of data can be passed in several iteration, where the first call 
+        should have an array of all target variables.
+
+
+        Parameters
+        ----------
+        X : (sparse) array-like, shape = [n_samples, n_features]
+            Data.
+
+        y : array-like, shape = [n_samples]
+            Multi-class targets.
+
+        Returns
+        -------
+        self
+        """
+        if self._first_partial_run:
+            if classes is None:
+                raise ValueError("Must contain all classes of target")
+            self._first_partial_run = False
+            self.classes_ = np.unique(classes)
+            self.n_classes = self.classes_.shape[0]
+            self.estimators_ = [clone(self.estimator) for i in
+                                range(self.n_classes *
+                                (self.n_classes-1) // 2)]
+
+        y = np.asarray(y)
+        check_consistent_length(X, y)
+        self.estimators_ = Parallel(n_jobs=self.n_jobs)(
+            delayed(_partial_fit_ovo_binary)(
+                self.estimators_.pop(0), X, y, self.classes_[i],
```

Comment:
I am a bit uncomfortable with the "popping" of the estimators. Is it really required? I would rather have something like:

<pre>
...
for estimators, (class1, class2)
in itertools.izip(self.estimators, itertools.product(self.classes_, self.classes_))
</pre>